test/lib/ansible_test/_data/requirements/constraints.txt

```
@@ -1,7 +1,7 @@
 # do not add a cryptography or pyopenssl constraint to this file, they require special handling, see get_cryptography_requirements in python_requirements.py
 # do not add a coverage constraint to this file, it is handled internally by ansible-test
 packaging < 21.0 ; python_version < '3.6' # packaging 21.0 requires Python 3.6 or newer
-paramiko < 2.9.0 # paramiko 2.9.0+ requires changes to the paramiko_ssh connection plugin to work with older systems
+# paramiko < 2.9.0 # paramiko 2.9.0+ requires changes to the paramiko_ssh connection plugin to work with older systems
```

I'm not exactly sure how to address vyos, since those hosts need to set `ansible_paramiko_use_rsa_sha2_algorithms` to `False`.

Seems like a fair amount of plumbing might need added to allow setting this in the inventory

---

I'd make the following change temporarily, so CI is testing against a newer paramiko version. Otherwise the older version that's preinstalled will be used instead. This should expose any compatibility issues with the new version.

```suggestion
paramiko >= 2.9.0 # HACK: force paramiko upgrade to test changes against newer versions
```